app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -344,4 +356,42 @@ def stale_cache?(last_refresh_time)
       last_refresh_time > current_val[:created_at]
     end
   end
+
+  private
+
+  #
+  # Checks if IP address to host name resolving is enabled.
+  #
+  # @return [Boolean] `true` if host name resolving is enabled in the configuration, `false` otherwise.
+  #
+  def resolve_ip_addresses?
+    ::Settings.ems.ems_redhat.resolve_ip_addresses
+  end
+
+  #
+  # Tries to convert the given IP address into a host name, doing a reverse DNS lookup if needed. If it
+  # isn't possible to find the host name the original IP address will be returned, and a warning will be
+  # written to the log.
+  #
+  # @param address [String] The IP address.
+  # @return [String] The host name.
+  #
+  def resolve_ip_address(address)
+    require 'resolv'
```

Comment:
If there is a hard-and-fast rule, usually it's what we described here: put at the top of the file. 

There are specific exceptions, though; For example, in the ovirt_metrics gem I needed to write a custom adapter for dealing with an old version of Postgres on Ovirt databases that isn't supported by Rails 5 (what we run). As the adapter isn't required for Rails < 5, I require it inline based on which version of Rails you're running. I don't want the adapter required when it isn't necessary (it's very large and expensive to load)

Here it looks like you're adding some core functionality that should always run, so we'll consider it a true dependency that we'll specify at the beginning of the file.